OrbitGl/CaptureSerializer.cpp

```diff
@@ -49,35 +54,80 @@ ErrorMessageOr<void> CaptureSerializer::Save(const std::string& filename) {
 }
 
 void CaptureSerializer::Save(std::ostream& stream) {
-  cereal::BinaryOutputArchive archive(stream);
   std::basic_ostream<char> Stream(&GStreamCounter);
-  cereal::BinaryOutputArchive CountingArchive(Stream);
+  google::protobuf::io::OstreamOutputStream out_counting_stream(&Stream);
+  google::protobuf::io::CodedOutputStream coded_counting_output(
+      &out_counting_stream);
+
   GStreamCounter.Reset();
-  SaveImpl(CountingArchive);
+  SaveImpl(&coded_counting_output);
   PRINT_VAR(GStreamCounter.Size());
   GStreamCounter.Reset();
 
-  SaveImpl(archive);
+  google::protobuf::io::OstreamOutputStream out_stream(&stream);
+  google::protobuf::io::CodedOutputStream coded_output(&out_stream);
+  SaveImpl(&coded_output);
+}
+
+void WriteMessage(const google::protobuf::Message* message,
+                  google::protobuf::io::CodedOutputStream* output) {
+  uint32_t message_size = message->ByteSizeLong();
+  output->WriteLittleEndian32(message_size);
+  message->SerializeToCodedStream(output);
 }
 
 //-----------------------------------------------------------------------------
-template <class T>
-void CaptureSerializer::SaveImpl(T& archive) {
+void CaptureSerializer::SaveImpl(
```

Comment:
In my opinion the final result of this should be to have a single protobuf message consisting of a sequence of small independent messages that we write in a unified manner to a file. Here we are porting over dumping very big and heterogeneous structures like before (just in protobuf form) which like before make backwards compatibility extremely hard to maintain.